### PR TITLE
Map the ChatPreviewStateChangeS2CPacket class

### PR DESCRIPTION
**Pull request by haykam821**

### CODE REVIEW

File: mappings/net/minecraft/network/listener/ClientPlayPacketListener.mapping [hunk 1]
```diff
@@ -205,3 +205,5 @@ CLASS net/minecraft/class_2602 net/minecraft/network/listener/ClientPlayPacketLi
 		ARG 1 packet
 	METHOD method_44075 onServerMetadata (Lnet/minecraft/class_7495;)V
 		ARG 1 packet
+	METHOD method_44286 onChatPreviewEnabled (Lnet/minecraft/class_7519;)V
```
Comment (apple502j) [inline mappings/net/minecraft/network/listener/ClientPlayPacketListener.mapping]: Rename?

Comment (haykam821) [reply mappings/net/minecraft/network/listener/ClientPlayPacketListener.mapping]: Fixed in 8c0672854874e8c9aae52a07e8508c73a763e814

**Comment by liach** [comment]
does this update preview content or toggles preview on/off? if it toggles, I'd call it ChatPreviewToggleS2CPacket.

**Comment by haykam821** [comment]
@liach The packet consists of a boolean which represents whether the chat preview should be enabled for the specific player.

File: mappings/net/minecraft/client/network/ServerInfo.mapping [hunk 2]
```diff
@@ -16,6 +16,7 @@ CLASS net/minecraft/class_642 net/minecraft/client/network/ServerInfo
 	FIELD field_3763 local Z
 	FIELD field_39355 LOGGER Lorg/slf4j/Logger;
 	FIELD field_39356 chatPreview Lnet/minecraft/class_642$class_7484;
+	FIELD field_39436 chatPreviewEnabled Z
```
Comment (apple502j) [inline mappings/net/minecraft/client/network/ServerInfo.mapping]: I've mapped them as `chatPreviewEnabledByServer` on #3164 - let me check which one's better

Comment (apple502j) [reply mappings/net/minecraft/client/network/ServerInfo.mapping]: This one is temporary, unlike other ones - guess both are bad.

Comment (haykam821) [reply mappings/net/minecraft/client/network/ServerInfo.mapping]: I merged the `temporaryChatPreviewState` naming from #3164 in b8392d70dd07c09b6f02c57150cd4c282bbe6240.

**Review by apple502j — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**